presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1365,35 +1380,39 @@ private VersionType toVersionType(TableVersionType type)
                 case VERSION:
                     return VersionType.VERSION;
             }
-            throw new SemanticException(NOT_SUPPORTED, type.toString(), "Table version type not supported.");
+            throw new SemanticException(NOT_SUPPORTED, "Table version type %s not supported." + type);
         }
         private Optional<TableHandle> processTableVersion(Table table, QualifiedObjectName name, Optional<Scope> scope)
         {
-            Expression asOfExpr = table.getTableVersionExpression().get().getAsOfExpression();
+            Expression stateExpr = table.getTableVersionExpression().get().getStateExpression();
             TableVersionType tableVersionType = table.getTableVersionExpression().get().getTableVersionType();
-            ExpressionAnalysis expressionAnalysis = analyzeExpression(asOfExpr, scope.get());
+            TableVersionOperator tableVersionOperator = table.getTableVersionExpression().get().getTableVersionOperator();
+            ExpressionAnalysis expressionAnalysis = analyzeExpression(stateExpr, scope.get());
             analysis.recordSubqueries(table, expressionAnalysis);
-            Type asOfExprType = expressionAnalysis.getType(asOfExpr);
-            if (asOfExprType == UNKNOWN) {
-                throw new PrestoException(INVALID_ARGUMENTS, format("Table version AS OF expression cannot be NULL for %s", name.toString()));
+            Type stateExprType = expressionAnalysis.getType(stateExpr);
+            if (tableVersionOperator == LESS_THAN) {
```

Comment:
Isn't this check redundant given the check in ConnectorMetadata?

Reply:
ConnectorMetadata check is for non-iceberg cases, it will not return error in BEFORE case. 
I will remove this check as part of my iceberg connector PR.

Reply:
Maybe an `NOT_SUPPORTED` error code? As the argument is indeed supported grammatically.